### DISCUSSION TITLE
Modelling menu

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#9780] [General]
The modelling menu is lagging behind.
I suggest it remains crucial that we work on it, but also that we give it sufficient thought.  
I am reasonably comfortable with our general approach, which is based (as is our describe menu) with a successive number of variables.
1. SPSS is interesting in the set of non-parametric dialogues it suggests, and with their structure.  We don't want the same structure, but it gives a good list of procedures that we do want.  And they fit into our categories in that some are for one variable, many are for 2 variables, and the "paired" ones are for 3 variables.
2. Much more important is the set of facilities in the **robustbase** package.  This is important and is an area that was not particularly evident as a large subject in the 1960s.  In this package there are two functions in particular, namely lmrob and glmrob for robust alternatives to lm and glm.  There are also predict.lmrob and predict.glmrob.  This all fits nicely for us!

I still feel that getting a good and effective modelling menu could well be one year from now.

**Comment by lilyclements** [comment]
From @volloholic in #3565:

"A full implementation of 'down from the mountain' modelling including bootstrap, multinomial and Bayesian options. This system should include everything needed to teach at any level up to a standard full undergraduate statistics degree.

In particular dialogs should exist which enable the user to start with the data and then get an appropriate model following a basic decision tree.
The basic decision tree should be written up and open educational resources should exist to facilitate the process."
